crates/hir/src/diagnostics.rs

```diff
@@ -115,14 +116,14 @@ diagnostics![
 
 #[derive(Debug)]
 pub struct BreakOutsideOfLoop {
-    pub expr: InFile<AstPtr<ast::Expr>>,
+    pub expr: InFile<AstPtr<Either<ast::Expr, ast::Pat>>>,
```

Comment:
```suggestion
    pub expr: InFile<ExprOrPatPtr>,
```
should work right? (and elsewhere in the file)